src/sources/dnstap/mod.rs

```diff
@@ -195,8 +203,20 @@ impl FrameHandler for DnstapFrameHandler {
 
         let mut log_event = LogEvent::default();
 
+        self.log_namespace.insert_vector_metadata(
+            &mut log_event,
+            self.timestamp_key().as_str(),
+            "ingest_timestamp",
+            chrono::Utc::now(),
+        );
+
         if let Some(host) = received_from {
-            log_event.insert(self.host_key().as_str(), host);
+            self.log_namespace.insert_vector_metadata(
+                &mut log_event,
+                self.host_key().as_str(),
+                "host",
+                host,
+            );
         }
 
         if self.raw_data_only {
```

Comment:
I'm not really sure if anything needs to be done here?

Is it ok to just insert the various data that dnstap parses from the message as before - or does this need to go into a separate namespace?

Reply:
You are correct. It's basically implementing a custom codec, and all of the data it "parses" goes directly on the event itself.